qa/integration-tests/src/test/java/io/camunda/zeebe/it/management/ControlledActorClockEndpointIT.java

```diff
@@ -0,0 +1,209 @@
+        zeebeClient.newDeployCommand().addProcessModel(process, "process.bpmn").send().join();
+    zeebeClient
+        .newCreateInstanceCommand()
+        .processDefinitionKey(
+            deployResult.getProcesses().stream()
+                .findFirst()
+                .orElseThrow()
+                .getProcessDefinitionKey())
+        .send()
+        .join();
+
+    // then - records are exported with a timestamp matching the pinned time
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              final var records = searchExportedRecords();
+              assertThat(records)
+                  .hasSize(10) // deploy + instance produces exactly 10 records
+                  .allMatch((record) -> record.getTimestamp() == pinnedAt.toEpochMilli());
+            });
+  }
+
+  @Test
+  void testOffsetTime() throws IOException, InterruptedException {
+    // given - Zeebe actor clock is offset
+    final var beforeRecords = Instant.now();
+    final var offsetZeebeTime = clockActuatorClient.offsetZeebeTime(Duration.ofHours(5));
+    final var process = Bpmn.createExecutableProcess().startEvent().endEvent().done();
+
+    // when - producing records
+    final var deployResult =
+        zeebeClient.newDeployCommand().addProcessModel(process, "process.bpmn").send().join();
+    zeebeClient
+        .newCreateInstanceCommand()
+        .processDefinitionKey(
+            deployResult.getProcesses().stream()
+                .findFirst()
+                .orElseThrow()
+                .getProcessDefinitionKey())
+        .send()
+        .join();
+
+    // then - records are exported with a timestamp matching the offset time
+    Awaitility.await()
+        .untilAsserted(
+            () -> {
+              final var records = searchExportedRecords();
+              assertThat(records)
+                  .hasSize(10)
+                  .allSatisfy(
+                      (record) -> {
+                        final var timestamp = Instant.ofEpochMilli(record.getTimestamp());
+                        final var afterRecords = Instant.now();
+                        assertThat(timestamp)
+                            .isBefore(afterRecords.plus(offsetZeebeTime))
+                            .isAfter(beforeRecords.plus(offsetZeebeTime));
+                      });
+            });
+  }
+
+  private List<AbstractRecord<?>> searchExportedRecords() throws IOException, InterruptedException {
+    final var uri =
+        URI.create(
+            String.format(
+                "http://%s/%s*/_search",
+                elasticsearchContainer.getHttpHostAddress(), INDEX_PREFIX));
+    final var request = HttpRequest.newBuilder(uri).method("POST", BodyPublishers.noBody()).build();
+    final var response = httpClient.send(request, BodyHandlers.ofInputStream());
+    final var result = mapper.readValue(response.body(), EsSearchResponseDto.class);
+    if (result.esDocumentsWrapper == null) {
+      return null;
```

Comment:
:upside_down_face:  Out of habit I would try to avoid return null unless you explicitly want to deal with null, and instead return `Collections.emptyList()` just in case. I know here it doesn't make much difference, but returning null has different semantics.